test/fork/one-time-partial-claims.js

```diff
@@ -0,0 +1,406 @@
+
+    await claims.submitPartialClaim(coverId, claimAmount, { from: coverOwner });
+
+    {
+      const coverInfo = await tokenController.coverInfo(coverId);
+      bnEqual(coverInfo.claimCount, 2);
+      assert(coverInfo.hasOpenClaim);
+      assert(!coverInfo.hasAcceptedClaim);
+      bnEqual(coverInfo.requestedPayoutAmount, claimAmount);
+    }
+  });
+
+  it('accept claim for cover 7160', async function () {
+    const { claims, claimsReward, claimsData, quotationData, tokenController, pool } = this;
+
+    const coverId = 7160; // FTX
+    const claimId = (await claimsData.actualClaimLength()) - 1;
+
+    await claims.submitCAVote(claimId, 1, { from: this.voters[0] });
+
+    // fast forward 72h
+    await time.increase(72 * 3600);
+
+    const coverAmount = await quotationData.getCoverSumAssured(coverId);
+    const coverOwner = await quotationData.getCoverMemberAddress(coverId);
+    const claimAmount = coverAmount.divn(4).muln(3);
+    const payoutAmount = ether(claimAmount.toString());
+
+    const coverOwnerBalanceBefore = toBN(await web3.eth.getBalance(coverOwner));
+    const poolBalanceBefore = toBN(await web3.eth.getBalance(pool.address));
+
+    const receipt = await claimsReward.closeClaim(claimId);
+    receipt.logs.forEach(log => {
+      console.log(log.event, filterArgsKeys(log.args));
+    });
+
+    const coverOwnerBalanceAfter = toBN(await web3.eth.getBalance(coverOwner));
+    const poolBalanceAfter = toBN(await web3.eth.getBalance(pool.address));
+
+    console.log('Cover owner ETH balance before:', coverOwnerBalanceBefore.toString() / 1e18);
+    console.log('Cover owner ETH balance after:', coverOwnerBalanceAfter.toString() / 1e18);
+
+    console.log('Pool ETH balance before:', poolBalanceBefore.toString() / 1e18);
+    console.log('Pool ETH balance after:', poolBalanceAfter.toString() / 1e18);
+
+    bnEqual(coverOwnerBalanceAfter.sub(coverOwnerBalanceBefore), payoutAmount);
+    bnEqual(poolBalanceBefore.sub(poolBalanceAfter), payoutAmount);
+
+    {
+      const coverInfo = await tokenController.coverInfo(coverId);
+      bnEqual(coverInfo.claimCount, 2);
+      assert(!coverInfo.hasOpenClaim);
+      assert(coverInfo.hasAcceptedClaim);
+      bnEqual(coverInfo.requestedPayoutAmount, claimAmount);
+    }
+
+    await expectRevert(
+      claims.submitPartialClaim(coverId, claimAmount, { from: coverOwner }),
+      'TokenController: Max claim count exceeded',
+    );
+
+    await expectRevert(
+      claims.submitClaim(coverId, { from: coverOwner }),
+      'TokenController: Max claim count exceeded',
+    );
+  });
+
+  it('processes pending actions', async function () {
+    const { pooledStaking } = this;
+    while (await pooledStaking.hasPendingActions()) {
+      await pooledStaking.processPendingActions(100);
+    }
+  });
+
+  it('submits full claim for cover 7907', async function () {
+    const { claims, quotationData, tokenController } = this;
+
+    const coverId = 7907; // FTX
```

Comment:
in retrospect a parametrized function for claim submission would have made these 2 test cases easier to read